### ISSUE TITLE
[Bug][DropDownList] Keyboard navigation skips item without label

### ISSUE THREAD

**Issue by filipKovachev** [telerik/kendo-react#2251]
## I'm submitting a...

* Bug report 

## Current behavior
Currently, the dropdowns act rather bizarre when navigating with the arrow keys over an item without a label. In some cases, the item without a label is skipped when navigating up and if you go down across the item it does not work which is inconsistent. This behavior can be observed in the ComboBox, DropDownList and MultiSelect

## Expected behavior
The keyboard navigation should be consistent and enable the user to traverse the items without labels.

## Minimal reproduction of the problem with instructions
1. Open this StackBlitz: https://stackblitz.com/edit/react-sgqlnn-vzreua?file=app%2Fmain.tsx
2. Navigate to the first item by pressing the down arrow, and notice you can't reach the last item. When going down the item is skipped, when going up with the arrow key it works. When pressing the arrow up, focus is skipped between the first and second item

Reported in Ticket ID: 1651564

**Comment by Sasho32** [comment]
The fix is available in v14.4.1-develop.14.
